ml-agents/mlagents/trainers/components/reward_signals/gail/model.py

```diff
@@ -228,7 +231,35 @@ def create_network(self) -> None:
         self.discriminator_score = tf.reshape(
             self.policy_estimate, [-1], name="GAIL_reward"
         )
-        self.intrinsic_reward = -tf.log(1.0 - self.discriminator_score + 1e-7)
+        self.intrinsic_reward = -tf.log(1.0 - self.discriminator_score + EPSILON)
+
+    def compute_gradient_penalty(self) -> tf.Tensor:
```

Comment:
What is the performance improvement from this? Faster/more stable convergence?

Reply:
Faster convergence, esp. for Crawler. I'm seeing about 25% less steps required with PPO. TBH a large motivation for this is for SAC, where without GP the discriminator will overfit very quickly - GAIL + SAC doesn't work at all without it.

Reply:
![Plots for Crawler + PPO](https://user-images.githubusercontent.com/5085265/61662684-c1956f00-ac83-11e9-8e4d-e1f6445b7ffa.png)

Reply:
The squiggly line does not lie.